### PR TITLE
Doc update: Concatenated gzip data handling

### PR DESCRIPTION
**Pull request by smtakeda**
@shazow I'm not sure if adding a page for concatenated gzip is ok. Please take a look and let me know. thanks.

### CODE REVIEW

File: docs/recipes.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+--------------------------------------------
+
+By default, urllib3 checks ``Content-Encoding`` header in HTTP response and decodes the data in ``gzip`` or ``deflate`` transparently. If ``Content-Encoding`` is not either of them, however, you will have to decode data in your application.
+
+This recipe shows how to decode data in the concatenated gzip format where multiple gzipped data chunks are concatenated in HTTP response. 
+
+.. doctest ::
+
+    import zlib
+    import requests
```
Comment (shazow) [inline docs/recipes.rst]: urllib3 things for urllib3, requests things for requests. :)

**Comment by shazow** [comment]
Looks fantastic, thank you! Only one comment, please keep it to just urllib3-related recipes. We have lots of projects that are frontends for urllib3 and requests is just one of them.

**Comment by smtakeda** [comment]
Thanks for the review and feedback. I'll update it by tomorrow.

On Tue, Jan 26, 2016 at 11:41 AM, Andrey Petrov <notifications@github.com>
wrote:

> Looks fantastic, thank you! Only one comment, please keep it to just
> urllib3-related recipes. We have lots of projects that are frontends for
> urllib3 and requests is just one of them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/shazow/urllib3/pull/796#issuecomment-175194950>.
>



-- 
Shige Takeda
smtakeda@gmail.com

**Comment by shazow** [comment]
No problem, ping back when it's ready. :)

File: docs/recipes.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
+Recipes
+=======
+
+This page includes a collection of recipes in the urlib3 cookbook.
+
+Decode HTTP Response Data in Concatenated Gzip Format
+--------------------------------------------
```
Comment (Lukasa) [inline docs/recipes.rst]: Can we have this underline run the whole length of the header?

**Comment by smtakeda** [comment]
done the update.

**Comment by shazow** [comment]
That's excellent, thank you @smtakeda! :custard:

**Pull request merged**